In [1]:
from IPython.core.display import HTML
styles = open( '../styles/custom.css', 'r' ).read()
HTML( styles )

In [2]:
from sklearn.datasets import load_diabetes
from sklearn import linear_model
import numpy as np

In [3]:
diabetes = load_diabetes()

In [4]:
# Use only one feature
# diabetes_X = diabetes.data[ :, np.newaxis ]
# diabetes_X_temp = diabetes_X[ :, :, 2 ]

# Split the data into training/testing sets
diabetes_X_train = diabetes.data[ : -20 ]
diabetes_X_test = diabetes.data[ -20 : ]

# Split the targets into training/testing sets
diabetes_y_train = diabetes.target[ : -20 ]
diabetes_y_test = diabetes.target[ -20 : ]

regr = linear_model.LinearRegression()

regr.fit( diabetes_X_train, diabetes_y_train )

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1, normalize=False)

In [5]:
regr.coef_, regr.intercept_

(array([  3.03499549e-01,  -2.37639315e+02,   5.10530605e+02,
          3.27736980e+02,  -8.14131709e+02,   4.92814588e+02,
          1.02848452e+02,   1.84606489e+02,   7.43519617e+02,
          7.60951722e+01]), 152.76430691633442)

$$
  diabetes = 152.8 + (3.0 \times 10^{-1} age )- (2.4 \times 10^2 sex )
  + (5.1 \times 10^2 bmi) + \cdots
  + (7.6 \times 10^1 glu)
$$

In [6]:
from sklearn.feature_selection import f_regression
f_regression( diabetes_X_train, diabetes_y_train, center = False )

(array([  3.64928159,   0.08105158,  32.82999821,  18.01931084,
          3.24920652,   2.04742062,  15.14471777,  16.91054201,
         29.15660548,  11.81705708]),
 array([  5.67725157e-02,   7.76017361e-01,   1.91831160e-08,
          2.69243707e-05,   7.21730902e-02,   1.53206135e-01,
          1.15726921e-04,   4.71315638e-05,   1.11803285e-07,
          6.45330233e-04]))

age sex bmi map tc ldl hdl tch ltg glu y